package-scripts/datadog-agent/preinstall

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh
+
+INSTALL_DIR=/opt/datadog-agent
+LOG_DIR=/var/log/datadog
+APP_DIR="/Applications/Datadog Agent.app"
+DD_COMMAND="/opt/datadog-agent/bin/datadog-agent"
+CONF_DIR="/opt/datadog-agent/etc"
+
+# We log stdout & stderr of this script
+LOG_FILE=$LOG_DIR/preinstall.log
+mkdir -p $LOG_DIR
+exec > $LOG_FILE 2>&1
+
+# OS/Distro Detection
+DISTRIBUTION=$(grep -Eo "(Debian|Ubuntu|RedHat|CentOS|openSUSE|Amazon)" /etc/issue 2>/dev/null || uname -s)
+
+if [ $DISTRIBUTION = "Darwin" ]; then
+  if [ -e "$CONF_DIR/datadog.conf" ]; then
+    echo "# State at the beginning"
+    echo "## Agent version"
+    grep AGENT_VERSION $INSTALL_DIR/agent/config.py || echo "No config file"
+    echo "## $INSTALL_DIR"
+    ls -al $INSTALL_DIR || echo "No agent installed"
+    echo "## $APP_DIR/Contents/Resources"
+    ls -al "$APP_DIR/Contents/Resources" || echo "No app installed"
+
+    echo '# Stop old agent'
+    $DD_COMMAND stop || true
+
+    echo '# Stop old GUI'
+    kill `ps aux | grep 'Datadog Agent.app' | grep -v grep  | awk '{ print $2 }'` || true
+
+    echo '# Save old conf'
+    mkdir -vp /tmp/{conf,checks}.d
+    rm -vf /tmp/{conf,checks}.d/* /tmp/datadog.conf
```

Comment:
Couldn't this line be a problem if someone had the (bad) idea to split some components of his checks into different files stored in a subfolder ?